File: Programming-languages-and-laboratory/Python/Exercises/29a_compito_java_20190411/GestionePalestra.py
from Dipendente import Dipendente
from Trainer import Trainer
from Nutrizionista import Nutrizionista
from Cliente import Cliente
from Servizio import Servizio


class GestionePalestra:

    def main():

        dipendenti = []
        clienti = []
        cod_to_dip = {}

        try:
            f = open("dipendenti.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice = int(tok[0])
                tipo = tok[1]
                nome = f.readline().strip()

                line = f.readline().strip()
                tok = line.split()
                telefono_ore = int(tok[0])

                if tipo == "trainer":
                    costo = float(tok[1])
                    specialita = f.readline().strip()

                    tr = Trainer(codice, tipo, nome, telefono_ore, costo, specialita)
                    dipendenti.append(tr)
                    cod_to_dip[codice] = tr
                else:
                    medico = bool(tok[1])
                    app_settimanali = int(tok[2])
                    costo = float(tok[3])

                    nut = Nutrizionista(codice, tipo, nome, telefono_ore, medico, app_settimanali, costo)
                    dipendenti.append(nut)
                    cod_to_dip[codice] = nut

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        try:
            f = open("clienti.txt", "r")
            line = f.readline().strip()

            while line != "":
                codice_cliente = int(line)
                nome_cliente = f.readline().strip()

                cl = Cliente(codice_cliente, nome_cliente)
                clienti.append(cl)

                line = f.readline().strip()

                while line != "":
                    tok = line.split()
                    codice_dipendente = int(tok[0])
                    ore_servizio = float(tok[1])
                    d = cod_to_dip[codice_dipendente]
                    d.incrementa_numero_servizi()
                    serv = Servizio(codice_dipendente, ore_servizio)
                    cl.add_servizio(serv, (d.get_costo() * ore_servizio))

                    line = f.readline().strip()

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("nome, codice, tipo, ore settimanali, specialità, telefono, medico, app.sett., costo orario")

        for d in dipendenti:
            print(d)
        print("")

        for c in clienti:
            print(c)
        print("")

        numero_servizi = 0
        dip_max_servizi = "boh"
        for dip in dipendenti:
            if dip.get_numero_servizi() > numero_servizi:
                dip_max_servizi = dip.get_nome()
                numero_servizi = dip.get_numero_servizi()

        print(dip_max_servizi + " " + str(numero_servizi))


if __name__ == "__main__":
    GestionePalestra.main()

File: II - Linguaggi di programmazione e laboratorio/Esame/Java/29_compito_java_2019_04_11a/GestionePalestra.java
import java.io.*;
import java.util.*;

public class GestionePalestra {

	static List<Dipendente> dipendenti = new LinkedList<Dipendente>();
	static List<Cliente> clienti = new LinkedList<Cliente>();
	static Map<Integer, Dipendente> codDip = new HashMap<Integer, Dipendente>();
	static Map<Integer, Integer> codNServ = new HashMap<Integer, Integer>();

	public static void main(String[] args) {

		// PUNTO 1: memorizzazione progetti
		try {
			BufferedReader br = new BufferedReader(new FileReader("dipendenti.txt"));
			String line = br.readLine();
			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int cod = Integer.parseInt(tok.nextToken());
				String tipo = tok.nextToken();
				String nome = br.readLine();
				line = br.readLine();
				if (tipo.equals("trainer")) {
					tok = new StringTokenizer(line);
					int ore = Integer.parseInt(tok.nextToken());
					double costo = Double.parseDouble(tok.nextToken());
					line = br.readLine();
					String spec = line;
					Dipendente d = new Trainer(cod, nome, ore, spec, costo);
					dipendenti.add(d);
					codDip.put(cod, d);
					codNServ.put(cod, 0);
				} else {
					tok = new StringTokenizer(line);
					int tel = Integer.parseInt(tok.nextToken());
					boolean medico = Boolean.parseBoolean(tok.nextToken());
					int app = Integer.parseInt(tok.nextToken());
					double costo = Double.parseDouble(tok.nextToken());
					Dipendente d = new Nutri(cod, nome, tel, medico, app, costo);
					dipendenti.add(d);
					codDip.put(cod, d);
					codNServ.put(cod, 0);
				}
				line = br.readLine();
			}
			br.close();
		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}
		// PUNTO 2: lettura di tutti i clienti

		try {
			BufferedReader br = new BufferedReader(new FileReader("clienti.txt"));
			String line = br.readLine();
			while (line != null) {
				int codcli = Integer.parseInt(line);
				String nomeCog = br.readLine();
				Cliente c = new Cliente(codcli, nomeCog);
				clienti.add(c);
				line = br.readLine();
				while (line != null && !line.equals("")) {
					StringTokenizer tok = new StringTokenizer(line);
					int cod = Integer.parseInt(tok.nextToken());
					double ore = Double.parseDouble(tok.nextToken());
					Dipendente dip = codDip.get(cod);
					c.addServizio(dip, ore);
					int current = codNServ.get(cod);
					codNServ.put(cod, current + 1);
					line = br.readLine();
				}
				line = br.readLine();
			}
			br.close();
		} catch (IOException e) {
			System.err.println(e);
		}

		// PUNTO 3: stampa di tutti i dipendenti
		System.out
				.println("nome, codice, tipo, ore settimanali, specialità, telefono, medico, app.sett., costo orario");
		for (Dipendente d : dipendenti)
			System.out.println(d);
		System.out.println();
		// //PUNTO 4: stampa clienti
		for (Cliente c : clienti)
			System.out.println(c);

		// PUNTO 5: stampe del dipendente che ha fatto più servizi
		System.out.println();
		int max = -1;
		String nome = null;
		for (Dipendente d : dipendenti) {
			int count = codNServ.get(d.getCod());
			if (count > max) {
				max = count;
				nome = d.getNome();
			}
		}
		System.out.println(nome + " " + max);

	}
}